buildSrc/src/main/java/com/google/firebase/gradle/plugins/GmavenVersionChecker.kt

```diff
@@ -0,0 +1,56 @@
+package com.google.firebase.gradle.plugins
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.TaskAction
+
+abstract class GmavenVersionChecker : DefaultTask() {
+
+  @get:Input abstract val groupId: Property<String>
+
+  @get:Input abstract val artifactId: Property<String>
+
+  @get:Input abstract val latestReleasedVersion: Property<String>
+
+  @get:Input abstract val version: Property<String>
+
+  @TaskAction
+  fun run() {
```

Comment:
The check verifies if 2 different fields are correct

- `latestReleasedVersion` should match the `latest` value in gmaven
- `version` should be one above `latestReleasedVersion`

By only throwing after verifying both fields we can have a better error message that shows both error, if present.

Reply:
With the adjustment to printing an error then including instructions for desired state, I think it might be less confusing to only print a single error with instructions rather than print several related errors, so the release operator doesn't assume there are multiple problems to fix. WDYT?

Reply:
sgtm